app/screens/home/bottomTabs/Articles.tsx

```diff
@@ -421,76 +448,169 @@ const Articles = ({ route, navigation }: any): any => {
 
 
   const onFilterArrayChange = (newFilterArray: any): any => {
-
+    console.log('filterarray data', newFilterArray)
     setFilterArray(newFilterArray)
   }
+  const customSortFunction = (a: any, b: any, queryText: any): any => {
+    // Check if a and b are valid strings
+    
+    if (typeof a !== 'string' || typeof b !== 'string') {
+      return 0; // Return 0 if either a or b is not a string
+    }
+
+    // Convert a and b to lowercase strings for case-insensitive comparison
+    const aLower = a.toLowerCase();
+    const bLower = b.toLowerCase();
+
+    // Check if aLower and bLower are valid strings after conversion
+    if (typeof aLower !== 'string' || typeof bLower !== 'string') {
+      return 0; // Return 0 if either aLower or bLower is not a string
+    }
+
+    // Find the index of queryText in aLower and bLower
+    const indexA = aLower.indexOf(queryText.toLowerCase());
+    const indexB = bLower.indexOf(queryText.toLowerCase());

```

Comment:
We need to test this logic with big long article to make sure performance is not a concern.